src/linter/ui5Types/SourceFileLinter.ts

```diff
@@ -1185,8 +1186,67 @@ export default class SourceFileLinter {
 		].includes(nodeType);
 	}
 
+	analyzeExportedValuesByLib(node: ts.PropertyAccessExpression | ts.ElementAccessExpression) {
+		if (!ts.isElementAccessExpression(node) &&
+			node.name?.kind !== ts.SyntaxKind.Identifier) {
+			return;
+		}
+
+		const extractVarName = (node: ts.PropertyAccessExpression | ts.ElementAccessExpression) => {
+			const nodeName = ts.isPropertyAccessExpression(node) ?
+					node.name.getText() :
+					node.argumentExpression.getText();
+
+			return nodeName.replaceAll("\"", "");
+		};
+
+		let exprNode = node.expression;
+		const namespace: string[] = [];
+		while (ts.isPropertyAccessExpression(exprNode) ||
+			ts.isElementAccessExpression(exprNode)) {
+			namespace.unshift(extractVarName(exprNode));
+			exprNode = exprNode.expression;
+		}
+		const exprType = this.checker.getTypeAtLocation(exprNode);
+		const potentialLibImport = exprType.symbol?.getName().replaceAll("\"", "") ?? "";
+
+		// Checks if the left hand side is a library import.
+		// It's sufficient just to check for "/library" at the end of the string by convention
+		if (!potentialLibImport.endsWith("/library")) {
+			return;
+		}
+
+		const varName = extractVarName(node);
+		const moduleName = [
+			potentialLibImport.replace("/library", ""),
+			...namespace,
+			varName,
+		].join("/");
+
+		// Check if the module is registered within ambient modules
+		const ambientModules = this.checker.getAmbientModules();
+		const libAmbientModule = ambientModules.find((module) =>
+			module.name === `"${potentialLibImport}"`);
+		const isRegisteredAsUi5Module = ambientModules.some((module) =>
+			module.name === `"${moduleName}"`);
+
+		if (isRegisteredAsUi5Module && !libAmbientModule?.exports?.has(varName as ts.__String)) {
+			this.#reporter.addMessage(MESSAGE.NO_EXPORTED_VALUES_BY_LIB, {
+				module: [
+					exprNode?.getText(),
+					...namespace,
+					varName,
+				].join("/"),
+				namespace: moduleName,
```

Comment:
I would have expected to have the `moduleName` passed as `module` and to have the local access (e.g. `unifiedLibrary.CalendarDayType`) passed as `namespace`.

Reply:
Done